src/tree/fleetTreeItem.ts

```diff
@@ -0,0 +1,97 @@
+import { AzExtParentTreeItem, AzExtTreeItem, IActionContext } from "@microsoft/vscode-azext-utils";
+import { CloudExplorerV1 } from "vscode-kubernetes-tools-api";
+import { assetUri } from "../assets";
+import { DefinedFleetMemberWithGroup, DefinedResourceWithGroup } from "../commands/utils/azureResources";
+import { SubscriptionTreeNode } from "./subscriptionTreeItem";
+import { createClusterTreeNode } from "./aksClusterTreeItem";
+import { parseResource } from "../azure-api-utils";
+
+// The de facto API of tree nodes that represent individual AKS clusters.
+// Tree items should implement this interface to maintain backward compatibility with previous versions of the extension.
+export interface FleetTreeNode {
+    readonly nodeType: "fleet";
+    readonly resourceType: string;
+    readonly armId: string;
+    readonly name: string;
+    readonly subscriptionTreeNode: SubscriptionTreeNode;
+    readonly subscriptionId: string;
+    readonly resourceGroupName: string;
+    readonly fleetResource: DefinedResourceWithGroup;
+    addCluster(clusters: DefinedResourceWithGroup[]): void;
+}
+
+export function createFleetTreeNode(
+    parent: AzExtParentTreeItem & SubscriptionTreeNode,
+    subscriptionId: string,
+    fleetResource: DefinedResourceWithGroup,
+): FleetTreeItem {
+    return new FleetTreeItem(parent, subscriptionId, fleetResource);
+}
+
+class FleetTreeItem extends AzExtParentTreeItem implements FleetTreeNode {
+    public readonly subscriptionTreeNode: SubscriptionTreeNode;
+    public readonly armId: string;
+    public readonly resourceGroupName: string;
+    public readonly name: string;
+
+    constructor(
+        parent: AzExtParentTreeItem & SubscriptionTreeNode,
+        readonly subscriptionId: string,
+        readonly fleetResource: DefinedResourceWithGroup,
+    ) {
+        super(parent);
+
+        this.iconPath = assetUri("resources/fleet-tree-icon.png");
+        this.subscriptionTreeNode = parent;
+        this.id = `${this.fleetResource.name} ${fleetResource.resourceGroup}`;
+        this.armId = this.fleetResource.id;
+        this.resourceGroupName = fleetResource.resourceGroup;
+        this.name = this.fleetResource.name;
+    }
+
+    public readonly contextValue: string = `fleet.hub ${CloudExplorerV1.SHOW_KUBECONFIG_COMMANDS_CONTEXT}`;
```

Comment:
let's remove what's after "fleet.hub" so that there isn't any menu item on the fleet node for now